### PR TITLE
Newly created posts should be world readable! Also, if you fail to post

### PR DESCRIPTION
**Pull request by jfly**
a competition announcement, we still want it to be world readable.

### CODE REVIEW

**Comment by jfly** [comment]
@timhabermaas, could you take a look?

File: WcaOnRails/app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -73,7 +74,10 @@ def destroy
     #  | 2014 |
     #  +------+
     #  1 row in set, 1 warning (0.00 sec)
-    world_readable_posts = Post.where(world_readable: true)
-    world_readable_posts.find_by_slug(params[:id]) || world_readable_posts.find_by_id!(params[:id])
+    post = Post.find_by_slug(params[:id]) || Post.find_by_id(params[:id])
+    if !post || !post.world_readable
+      raise ActiveRecord::RecordNotFound.new("Couldn't find post")
+    end
+    post
```
Comment (timhabermaas) [inline WcaOnRails/app/controllers/posts_controller.rb]: I think these 5 lines could be written as:

```ruby
Post.where(world_readable: true).where('slug = ? OR id = ?', params[:id], params[:id]).first!
```

`first!` raises the `RecordNotFound` and doesn't return `nil` (like `first` does).

**Comment by timhabermaas** [comment]
LGTM! I just had one stylistic remark which isn't really important.

File: WcaOnRails/app/controllers/posts_controller.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/controllers/posts_controller.rb]: I wish =(. See https://github.com/cubing/worldcubeassociation.org/commit/3ea44abf2db0dca36d1dee7abe535dfa5560b23e

Comment (timhabermaas) [inline WcaOnRails/app/controllers/posts_controller.rb]: Argh. :(

Comment (jfly) [inline WcaOnRails/app/controllers/posts_controller.rb]: Weirdly enough, with your suggested query, our tests did pass. I've expanded the relevant test so we won't be tempted to make such a change in the future: https://github.com/jfly/worldcubeassociation.org/commit/fb46ed78739556c62d6b076f2c4cf351d14e1399.

**Comment by jfly** [comment]
Thanks Tim! Will merge up now.

**Pull request merged**